testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/TestListener.java

```diff
@@ -33,9 +40,37 @@ public void testIgnored(Description description) throws Exception {
         ignoredTests.add(JUnitTestRunner.fullName(description));
     }
 
+    private String readFile(String path) throws IOException {
+        File file = new File(path);
+        return FileUtils.readFileToString(file, "UTF-8");
+    }
+
     @Override
     public void testStarted(Description description) throws Exception {
-        times.put(JUnitTestRunner.fullName(description), System.nanoTime());
+        String fullTestName = JUnitTestRunner.fullName(description);
+        times.put(fullTestName, System.nanoTime());
+
+        String phase = readFile(MainAgent.tmpfile);
```

Comment:
Instead of reading from a tmp file, maybe we can just pass in values as properties, like we do for most other things.